### PR TITLE
Dependency bumps

### PR DESCRIPTION
**Pull request by stevejalim**
Resolves #13411: Bump black from 23.3.0 to 23.7.0 in /requirements
Resolves #13410: Bump django-allow-cidr from 0.6.0 to 0.7.1 in /requirements
Resolves #13409: Bump ruff from 0.0.277 to 0.0.278 in /requirements
Resolves #13408: Bump lxml from 4.9.2 to 4.9.3 in /requirements
Resolves #13406: Bump basket-client from 1.0.0 to 1.1.0 in /requirements
Resolves #13405: Bump sentry-sdk from 1.27.1 to 1.28.1 in /requirements
Resolves #13403: Bump pipdeptree from 2.9.4 to 2.10.1 in /requirements
Resolves #13402: Bump boto3 from 1.28.1 to 1.28.3 in /requirements
Resolves #13401: Bump django-cors-headers from 4.1.0 to 4.2.0 in /requirements
Resolves #13400: Bump cryptography from 41.0.1 to 41.0.2 in /requirements

Please check the Integration tests before merging https://github.com/mozilla/bedrock/actions/runs/5629616984

### CODE REVIEW

**Comment by stevejalim** [comment]
Marking a draft while I fix up some linting regressions

**Review by robhudson — APPROVED** [review]
r+wc

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -65,11 +65,11 @@ repos:
           - --comment-style
           - "|#|"
   - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: v0.0.272
+    rev: v0.0.277
```
[the remark this answers is not in the document]
Comment (stevejalim) [reply .pre-commit-config.yaml:68]: Will bump but I wonder if we can just pin a ceiling and let it auto-upgrade every time

File: pyproject.toml [hunk 2]
```diff
@@ -19,6 +19,7 @@ extend-exclude = ["__pycache__", "docs", "assets", "static", "bedrock/externalfi
 
 [tool.ruff.per-file-ignores]
 "bedrock/settings/__init__.py" = ["F405"]
+"bedrock/contentful/tests/data.py" = ["E501"]
```
[the remark this answers is not in the document]
Comment (stevejalim) [reply pyproject.toml:22]: Good catch - the in-file `noqa` didn't work, so I'll drop that and keep this

**Pull request merged**